Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredResponseAuditTransforms.java

```diff
@@ -0,0 +1,221 @@
+        participant.setUserIsRequestor(Boolean.TRUE);
+        return participant;
+
+    }
+    
+    @Override
+    protected AuditMessageType.ActiveParticipant getActiveParticipantDestination(NhinTargetSystemType target, 
+        boolean isRequesting, Properties webContextProperties, String serviceName) {
+
+        String url;
+        String ipOrHost;
+
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+
+            url = isRequesting ? getWebServiceUrlFromRemoteObject(getLocalHCIDNhinTarget(),
+                NhincConstants.PATIENT_DISCOVERY_DEFERRED_RESP_SERVICE_NAME): getWebServiceUrlFromRemoteObject(
+                    convertToNhinTarget(PRPAIN201306UV02Parser.getSenderHcid(getRequest())),
+                    NhincConstants.PATIENT_DISCOVERY_DEFERRED_RESP_SERVICE_NAME );
+        if (url != null) {
+            try {
+                participant.setUserID(url);
+                ipOrHost = new URL(url).getHost();
+                participant.setNetworkAccessPointID(ipOrHost);
+                participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(ipOrHost));
+            } catch (MalformedURLException ex) {
+                LOG.error("Couldn't parse the given NetworkAccessPointID as a URL: " + ex.getLocalizedMessage(), ex);
+                // The url is null or not a valid url; for now, set the user id to anonymous
+                participant.setUserID(AuditTransformsConstants.ACTIVE_PARTICIPANT_USER_ID_SOURCE);
+                // TODO: For now, hardcode the value to localhost; need to find out if this needs to be set
+                participant.setNetworkAccessPointTypeCode(AuditTransformsConstants.NETWORK_ACCESSOR_PT_TYPE_CODE_NAME);
+                participant.setNetworkAccessPointID(AuditTransformsConstants.ACTIVE_PARTICIPANT_UNKNOWN_IP_ADDRESS);
+            }
+        }
+        if (!isRequesting) {
+            participant.setAlternativeUserID(ManagementFactory.getRuntimeMXBean().getName());
+        }
+        participant.setUserIsRequestor(Boolean.FALSE);
+        participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_ROLE_CODE_DEST, null,
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_CODE_SYSTEM_NAME,
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_ROLE_CODE_DESTINATION_DISPLAY_NAME));
+
+        return participant;
+    }
+
+
+    
+    protected String getPDDeferredRequestInitiatorAdress() {
+        String ipOrHost;
+        try {
+            String hcid = PRPAIN201306UV02Parser.getReceiverHCID(getRequest());
+            ipOrHost = new URL(getWebServiceUrlFromRemoteObject(MessageGeneratorUtils.getInstance().
+                convertToNhinTargetSystemType(createNhinTargetCommunity(hcid)),
+                NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME)).getHost();
+            return ipOrHost;
```

Comment:
Never assign a variable, and then return in the next line.  This needs to be "return new URL(...snipped...).getHost()".